what4/src/What4/Protocol/SMTLib2.hs

```diff
@@ -912,3 +937,128 @@ shutdownSolver _solver p = do
   ec <- Process.waitForProcess (solverHandle p)
   stopHandleReader (solverStderr p)
   return (ec,txt)
+
+
+-----------------------------------------------------------------
+-- Checking solver version bounds
+
+mkChunks :: [Word] -> [Versions.VChunk]
+mkChunks = map ((:[]) . Versions.Digits)
+
+-- | The minimum (inclusive) version bound for a given solver.
+--
+-- The keys come from @'smtWriterName'@ in @'WriterConn'@.
+-- See also https://github.com/GaloisInc/crucible/issues/194
+solverMinVersions :: Map String Version
+solverMinVersions =
+  [ -- TODO: Why is this verion required?
```

Comment:
Yices 2.6.1 provided the `print-success` command line option, which causes yices to print an acknowledgement message after every successful command.  We require this option be set to ensure that we maintain synchronization with the solver.

The reason for this is that Yices will sometimes respond with `unsat` to an `assert` command if the context becomes "obviously" unsatisfiable.  In order to reliably detect this condition, we require all commands to respond with some acknowledgement.

Reply:
Okay, I'll put this in a comment once we decide on an overall approach. Thanks for explaining this in more detail, I only vaguely remembered our Mattermost conversation after two weeks away :)